app/models/ability.rb

```diff
@@ -9,6 +9,8 @@ def initialize(user)
         can :manage, :all
       else
         can :manage, :all
```

Comment:
I don't think you want non-admins being able to manage all.  You should remove that line and give non-admin users access to specific resources, like you did on the two lines below.  Scope their access to where the `user_id` or `id` matches the user being passed into the ability class.  Admin users should probably be able to manage everything.